recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,196 @@
-            "--disable-multilib",
-            "--disable-bootstrap",
-            "--with-system-zlib",
-            "--with-gmp=%s" % self.deps_cpp_info['gmp'].rootpath,
-            '--with-mpc=%s' % self.deps_cpp_info["mpc"].rootpath,
-            "--with-mpfr=%s" % self.deps_cpp_info["mpfr"].rootpath,
-            "--without-isl",
-            "--libdir=%s" % libdir,
-            '--with-pkgversion=%s' % pkgversion,
-            "--program-suffix=-%s" % self.version,
-            "--with-bugurl=%s" % bugurl
-        ]
-        if self.settings.os == "Macos":
-            xcrun = tools.XCRun(self.settings)
-            args.extend([
-                '--with-native-system-header-dir=/usr/include',
-                "--with-sysroot={}".format(xcrun.sdk_path)
-            ])
-        self._autotools.libs = []  # otherwise causes config.log to fail finding -lmpc
+    def configure(self):
         if self.settings.compiler in ["clang", "apple-clang"]:
-            # xgcc: error: unrecognized command-line option -stdlib=libc++
-            if self.settings.compiler.libcxx == "libc++":
-                self._autotools.cxx_flags.remove("-stdlib=libc++")
-            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
-                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+            # Can't remove this from cxxflags with autotools - so get rid of it
+            del self.settings.compiler.libcxx
+
+    def build_requirements(self):
+        if self.settings.os == "Linux":
+            # binutils recipe is broken for Macos, and Windows uses tools
+            # distributed with msys/mingw
+            self.tool_requires("binutils/2.38")
+        self.tool_requires("flex/2.6.4")
 
     def requirements(self):
         self.requires("mpc/1.2.0")
         self.requires("mpfr/4.1.0")
-        self.requires("gmp/6.2.0")
-        self.requires("zlib/1.2.11")
+        self.requires("gmp/6.2.1")
+        self.requires("zlib/1.2.13")
+        self.requires("isl/0.24")
 
-    def configure(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
-        if tools.cross_building(self.settings):
-            raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
+    def package_id(self):
+        del self.info.settings.compiler
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "gcc-%s" % self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def validate_build(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("GCC can't be built with MSVC")
+
+    def validate(self):
+        if self.info.settings.os == "Windows":
```

Comment:
```suggestion
        if self.settings.os == "Windows":
```

``validate()`` doesn't need to use ``self.info`` anymore, but instead using ``self.settings`` is recommended.

Reply:
I'm finding it really hard to understand when to use `self.info` and when not to. If we shouldn't use it in `validate` anymore, then I'd suggest the [migration guide](https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-validate-method) needs to be updated to reflect that because it still provides explicit guidance to use `self.info`.

Are there any resources that will help me understanding the difference between `self.{info,}.{settings,options}` and when to apply which?

For this specific PR - should I also change the usage in the `package_id()`, `generate()` and `package_info()` methods?

Reply:
Docs are hard ! There always our dated 🙄 

We've flipped flopped a bunch on this one...

Info is only for package ID, everything else should be settings is the baseline.

`self.info` is for the compatibility layer, thats when computing required packages building the graph. "This is what the package contains" or the "can I use this package" are answered by `package_id`... that is how I've understood it.  On the other hand the `self.settings` is the "use these actually settings from my profile" to do stuff.

Hopefully that helps :) I added a task to fix the docs

Reply:
Thanks @prince-chrismc that clears things up for me. I've updated the recipe so that `self.info` is only used in the `package_id()` method.